In [16]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [3]:
data=pd.read_csv('Costomer+OrderForm.csv')
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [10]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [11]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [12]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [13]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [14]:
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [20]:
result=chi2_contingency(obs)
result

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [21]:
stat_value = result[0]
pi_value = result[1]
df = result[2]
stat_value, pi_value, df

(3.858960685820355, 0.2771020991233135, 3)

Ans) As p-value(0.2771) is greater than alpha significance(0.05) we can conclude that - "We fail to reject null hypothesis i.e. All proportions are equal."